### PR TITLE
Rename Handler Nested & Collection

### LINKED ISSUE #8765: Review Handler.Container so that it only represents an immutable interface
**Jetty version(s)**
Jetty 12

**Description**
The Handler.Container interface should be an immutable interface (without `addHandler()` methods).

### PR DESCRIPTION
**Pull request by gregw**
Fixes #8765  #8759 . Part of #9072

There is now a Handler interface hierarchy:
 + Container is a Handler that has 1 or more contained Handlers.
 + Wrapper is a Container with only 1 handler and a setHandler method.
 + Collection is a Container with n handlers and a addHandler method

class are now:
 + Abstract implements Handler
 + AbstractContainer extends Abstract implements Container
 + BaseWrapper extends AbstractContainer implements Wrapper
 + Sequence extends AbstractContainer implements Collection

 Lots of other associated cleanups

### CODE REVIEW

**Comment by gregw** [comment]
@sbordet @joakime @janbartel Can I get a review on this ASAP as I really don't want this to go stale and have to do it again.

**Review by joakime — APPROVED** [review]
I'm good with this.

**Comment by sbordet** [comment]
@gregw sorry I don't like some parts of this PR.

In every other Jetty class, `Wrapper` is a class and it is used to "wrap" an instance of an interface, so we should stick with this semantic.

`BaseWrapper` is not a good name.

I would rather have `Wrapper` back as a class, and if we really need it use `Single` as the interface opposed to `Collection`.

"Single" is also used in Java's `Collections.singletonList()` to return a list of 1 element ("Singleton" cannot be used as it remembers the well-known pattern which has a different meaning than what we want here).

**Comment by gregw** [comment]
@sbordet how about Delegate rather than Single?

**Comment by gregw** [comment]
Actually, I don't mind `Singleton` as it is now just as much associated with `Collections.singletonXxx()` than the pattern.   Thus a `Wrapper` is a concrete implementation of a `Singleton` `Container`.

File: jetty-ee9/jetty-ee9-maven-plugin/src/main/java/org/eclipse/jetty/ee9/maven/plugin/ServerSupport.java [hunk 1]
```diff
@@ -68,11 +66,7 @@ public static void configureHandlers(Server server, List<ContextHandler> context
         if (contexts == null)
         {
             contexts = new ContextHandlerCollection();
-            Handler.Collection handlers = server.getDescendant(Handler.Collection.class);
-            if (handlers == null)
-                server.setHandler(new Handler.Collection(contexts, new DefaultHandler()));
-            else
-                handlers.addHandler(contexts);
```
Comment (basil) [inline jetty-ee9/jetty-ee9-maven-plugin/src/main/java/org/eclipse/jetty/ee9/maven/plugin/ServerSupport.java:75]: Causes #12207, also decreasing the accuracy of the Javadoc a few lines above, which (now incorrectly) states:

> Also put in a `DefaultHandler` so we get a nicer page than a 404 if we hit the root and the webapp's context isn't at root.

**Pull request merged**